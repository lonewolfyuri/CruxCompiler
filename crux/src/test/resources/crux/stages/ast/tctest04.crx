// function can't have void arguments

func foo(a:void) : void {
  return ::foo(a);
}

func main() : void {
  ::printInt(7);
}
