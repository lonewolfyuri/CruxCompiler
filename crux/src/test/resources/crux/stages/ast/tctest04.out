ast.DeclarationList(3,0)
  ast.FunctionDefinition(3,0)[Symbol(foo), [Symbol(a)]]
    ast.StatementList(4,2)
      ast.Return(4,2)
        ast.Call(4,9)[Symbol(foo)]
          ast.Dereference(4,15)
            ast.Name(4,15)[a]
  ast.FunctionDefinition(7,0)[Symbol(main), []]
    ast.StatementList(8,2)
      ast.Call(8,2)[Symbol(printInt)]
        ast.LiteralInt(8,13)[7]
